test_utils_helpers.py

```
@@ -0,0 +1,1451 @@
+            "database.primary.host",
+            "database.replicas"
+        ]
+        
+        for key in expected_keys:
+            assert key in result
+    
+    def test_flatten_with_numeric_and_boolean_keys(self):
+        """Test flattening with non-string keys"""
+        data = {
+            "config": {
+                1: "first_item",
```

Flake8 (7.2.0)

[error] dictionary key 1 repeated with different values

(F601)